announcer.rb

```diff
@@ -2,22 +2,37 @@
 
 require 'json'
 require 'net/http'
-require 'optparse'
+require 'octokit'
 require 'pry-byebug'
 
 module Announcer
   PVTA_API_URL = 'http://bustracker.pvta.com/InfoPoint/rest'
 
   CONFIG_FILE = 'config.json'
-  MISSING_TEXT_FILE = 'missing_messages.log'
+  MISSING_TEXT_FILE = 'missing_messages.tmp'
+  PRESENT_TEXT_FILE = 'present_messages.tmp'
   QUERY_STOPS_FILE = 'stop_ids.txt'
   DEPARTURES_CACHE_FILE = 'cached_departures.json'
   AUDIO_COMMAND = File.read('audio_command.txt').strip
   SPEECH_COMMAND = File.read('speech_command.txt').strip
+  GITHUB_TOKEN = File.read('github_token.txt').strip
 
   @interval = 5
   @query_stops = %w[71 72 73]
 
+  def announce_all
+    set_query_stops
+    soonest_departures(new_departures).each do |announcement|
+      make_announcement announcement.merge(options: { exclude_stop_name: true })
+    end
+    update_github_issues!
+    remove_temp_files!
+  end
+
+  def announcements_in_progress?
+    File.file?(MISSING_TEXT_FILE) || File.file?(PRESENT_TEXT_FILE)
+  end
```

Comment:
Given that the GitHub updating process adds some time, I didn't want a particularly long string of announcements to be interrupted halfway through by the following minute's announcements.

This would also resolve the issue where on-demand announcements might be interrupted by the automated ones.

Reply:
Hmm... I also wonder about calling out to the filesystem here. If it were me, I'd use some sort of semaphore variable.

Reply:
Right, but this isn't "am *I* already running?", this is "is another announcements process already running"?

Reply:
Ah, yes. Never mind. For some resone I was thinking about this as a long-running service type process rather than as a periodically called one-shot.